docs/docs/guides/evaluation/guardrails_and_monitors.md

```diff
@@ -5,139 +5,482 @@ import TabItem from '@theme/TabItem';
+
+async def generate_with_monitoring(prompt: str) -> str:
+    # Get both the result and tracking information
+    result, call = generate_text.call(prompt)
     
-    Args:
-        user_input: The user's input to process
-        
-    Returns:
-        str: Processed result if guardrail passes, fallback response if it fails
-    """
-    result, call = op.call(user_input)
-    evaluation = await call.apply_scorer(guardrail)
+    # Sample monitoring (only monitor 10% of calls)
+    if random.random() < 0.1:
+        # Monitor multiple aspects asynchronously
+        await call.apply_scorer(QualityScorer())
+        await call.apply_scorer(RelevanceScorer())
     
-    if evaluation.score < 0.5:
-        return handle_failed_guardrail(result)
     return result
 ```
 
-## Monitors
+## Implementation Details
 
-While guardrails provide active intervention, monitors offer passive evaluation and tracking of LLM operations. They are crucial for long-term quality assurance and system improvement.
+### The Scorer Interface
 
-### Implementation
+A scorer is a class that inherits from `Scorer` and implements a `score` method. The method receives:
+- `output`: The result from your function
+- Any input parameters matching your function's parameters
+
+Here's a comprehensive example:
 
 ```python
-async def monitored_operation(user_input: str, sampling_rate: float = 0.25) -> str:
-    """
-    Execute operation with monitoring.
-    
-    Args:
-        user_input: The input to process
-        sampling_rate: Percentage of operations to monitor (0.0 to 1.0)
+@weave.op()
+def generate_styled_text(prompt: str, style: str, temperature: float) -> str:
+    """Generate text in a specific style."""
+    return "Generated text in requested style..."
+
+class StyleScorer(Scorer):
+    @weave.op
+    def score(self, output: str, prompt: str, style: str) -> dict:
+        """
+        Evaluate if the output matches the requested style.
         
-    Returns:
-        str: Operation result
-    """
-    result, call = op.call(user_input)
-    
-    # Apply monitoring based on sampling rate
-    if random.random() < sampling_rate:
-        await call.apply_scorer(scorer)
+        Args:
+            output: The generated text (automatically provided)
+            prompt: Original prompt (matched from function input)
+            style: Requested style (matched from function input)
+        """
+        return {
+            "style_match": 0.9,  # How well it matches requested style
+            "prompt_relevance": 0.8  # How relevant to the prompt
+        }
+
+# Example usage
+async def generate_and_score():
+    # Generate text with style
+    result, call = generate_styled_text.call(
+        prompt="Write a story",
+        style="noir",
+        temperature=0.7
+    )
     
-    return result
+    # Score the result
+    score = await call.apply_scorer(StyleScorer())
+    print(f"Style match score: {score.score['style_match']}")
```

Comment:
`score.score['style_match']`  --> `score.result['style_match']`

Reply:
nice - got it